main/src/main/java/cgeo/geocaching/utils/OfflineTranslateUtils.java

```diff
@@ -0,0 +1,368 @@
+        if (!Settings.getTranslationTargetLanguage().isValid()) {
+            translationBox.setVisibility(View.GONE);
+            return;
+        }
+        final TextView note = translationBox.findViewById(R.id.description_translate_note);
+        final Button button = translationBox.findViewById(R.id.description_translate_button);
+
+        // add observer to listing language
+        cda.translationStatus.setSourceLanguageChangeConsumer(lng -> {
+            if (null == lng || Settings.getLanguagesToNotTranslate().contains(lng.getCode())) {
+                translationBox.setVisibility(View.GONE);
+            } else if (OfflineTranslateUtils.LANGUAGE_UNKNOWN.equals(lng.getCode())) {
+                button.setEnabled(false);
+                note.setText(R.string.translator_language_unknown);
+            } else {
+                button.setEnabled(true);
+                note.setText(cda.getResources().getString(R.string.translator_language_detected, lng.toString()));
+            }
+        });
+
+        button.setOnClickListener(v -> callback.run());
+        note.setOnClickListener(v -> OfflineTranslateUtils.showLanguageSelection(cda, cda.translationStatus::setSourceLanguage));
+        translationBox.setVisibility(View.VISIBLE);
+
+        // identify listing language
+        OfflineTranslateUtils.detectLanguage(translateText,
+                cda.translationStatus::setSourceLanguage,
+                error -> {
+                    button.setEnabled(false);
+                    note.setText(error);
+                });
+    }
+
+    public static class TranslationProgressHandler extends ProgressButtonDisposableHandler {
+        public TranslationProgressHandler(final AbstractActivity activity) {
+            super(activity);
+        }
+    }
+
+    public static class Language implements Comparable<Language> {
+        private final String code;
+
+        public Language(final String code) {
+            this.code = code;
+        }
+
+        public String getCode() {
+            return code;
+        }
+
+        public boolean isValid() {
+            return !code.isEmpty();
+        }
+
+        public String getDisplayName() {
+            return new Locale(code).getDisplayName();
+        }
+
+        @NonNull
+        @Override
+        public String toString() {
+            return getDisplayName() + " (" + getCode() + ")";
+        }
+
+        @Override
+        public boolean equals(final Object o) {
+            if (!(o instanceof Language)) {
+                return false;
+            }
+            return ((Language) o).getCode().equals(getCode());
+        }
+
+        @Override
+        public int hashCode() {
+            return code.hashCode();
+        }
+
+        @Override
+        public int compareTo(@NonNull final Language lng) {
+            return this.toString().compareTo(lng.toString());
+        }
+    }
+
+    public static class Status {
```

Comment:
You use Status as member type in an activity and thus have to save and recreate it there eg on screen orientation change. 

You might get in trouble here because of objects like MutableLiveData, TranslationProgressHandler and a Consumer because those are not serializable/parcelable. Better move those things to another class which can be recreated from a "status" object on activity (re)start

Reply:
I understand your point about restoring it. I disagree with the priority of it. 
Also I don't know how to do it and will not have the time to figure that out soonish. So if that is a requirement I'll have to abandon this PR for the next weeks.
Anyone is free to contribute that part of course.

Reply:
That's fair. Using the "Status" approach you have chosen I too don't see an obvious way to accomplish this. In other places I created "Activity helper"s to accomplish this (eg Image-Selektion and SAF), but this would mean a major refactoring.

I will leave decision on how to proceed to the dev team.